src/app/debug/table/table.component.html

```diff
@@ -1,113 +1,133 @@
-          <td
-            *ngFor="let metadataName of viewSettings.currentView.metadataNames"
-            [style.padding]="getTableSpacing()"
-            class="table-cell"
-            (click)="openReport(row.storageId)"
-            [title]="row[metadataName]">
-            {{ row[metadataName]|tableCellShortener }}</td>
-        </tr>
-      </tbody>
-    </table>
+      @switch (getAmountOfReportsSelected()) {
+        @case (1) {
+          <app-button text="Open In New Tab" (click)="openReportInTab()"></app-button>
+        }
+        @case (2) {
+          <app-button text="Compare" title="Compare" (click)="compareTwoReports()"></app-button>
+        }
+      }
+      <app-active-filters class="ml-2" [activeFilters]="currentFilters" (click)="toggleFilter()"></app-active-filters>
+    </div>
+    @if (!doneRetrieving) {
+      <div>
+        <mat-spinner mode="indeterminate" [diameter]="30"></mat-spinner>
+      </div>
+    }
+    @if (doneRetrieving) {
+      <div data-cy-debug="table" class="table-responsive" id="metadataTable">
+        <table class="table mb-0" matSort (matSortChange)="helperService.sortData($event, tableSettings.reportMetadata)" [style.font-size]="getFontSize()">
+          <thead id="table-header">
+            <tr>
+              <th class="table-row-checkbox header-padding">
+                <input data-cy-debug="selectAll" type="checkbox" class="vertical-middle" [checked]="allRowsSelected" (click)="selectAllRows()" title="Select all rows" [style.width]="getCheckBoxSize()" [style.height]="getCheckBoxSize()">
+              </th>
+              @for (header of viewSettings.currentView.metadataLabels; let i = $index; track header) {
+                <th class="table-id-cell header-cell header-padding"
+                    [attr.mat-sort-header]="i"
+                    [title]="header">
+                  {{ getShortenedTableHeaderNames(header) }}
+                </th>
+              }
+            </tr>
+          </thead>
+          <tbody data-cy-debug="tableBody">
+            @for (row of tableSettings.reportMetadata.slice(0, tableSettings.displayAmount); let i = $index; track row) {
+              <tr class="table-row"
+                  (click)="highLightRow(i);"
+                  [ngClass]="{'highlight': selectedRow === i}"
+                  [style.background-color]="getStatusColor(row)"
+                  [attr.data-cy-record-table-index]=i
+              >
+                <td class="table-row-checkbox" [style.padding]="getTableSpacing()">
+                  <input type="checkbox" [checked]="row.checked" (click)="toggleCheck(row)" [style.width]="getCheckBoxSize()" [style.height]="getCheckBoxSize()">
+                </td>
+                @for (metadataName of viewSettings.currentView.metadataNames; track metadataName) {
+                  <td
+                    [style.padding]="getTableSpacing()"
+                    class="table-cell"
+                    (click)="openReport(row.storageId)"
```

Comment:
This click on every table cell is weird, could you also change that?

Reply:
I moved both click methods to a separate method in the table component and called it from the tr element. I remoed the click event from the td element.